Rules/CommonScripts/ScoreboardRender.as

```diff
@@ -677,18 +677,32 @@ void onTick(CRules@ this)
 
 void onInit(CRules@ this)
 {
-	if(getNet().isServer())
-	{
-		this.set_u32("match_time", 0);
-		this.Sync("match_time", true);
-	}
+	onRestart(this);
 }
 
 void onRestart(CRules@ this)
 {
-	if(getNet().isServer())
+	if(isServer())
 	{
 		this.set_u32("match_time", 0);
 		this.Sync("match_time", true);
+		getMapName(this);
 	}
 }
+
+void getMapName(CRules@ this)
+{
+	CMap@ map = getMap();
+	if(map !is null)
+	{
+		string[] name = map.getMapName().split('/');	//Official server maps seem to show up as
+		string mapName = name[name.length() - 1];		//``Maps/CTF/MapNameHere.png`` while using this instead of just the .png
+
+		name.clear();
+		name = mapName.split('.'); //Now remove the .png
+		mapName = name[0];	//select the first part since thats all we care about
```

Comment:
In singleplayer sandbox the map name is `test.kaggen.cfg`. This would only get `test`, unless that is intentional. Use `name.substr(0, name.length() - 4)` instead.